iceoryx_posh/include/iceoryx_posh/internal/mepoo/segment_manager.inl

```diff
@@ -131,6 +131,31 @@ SegmentManager<SegmentType>::getSegmentInformationWithWriteAccessForUser(const p
     return segmentInfo;
 }
 
+template <typename SegmentType>
+inline typename SegmentManager<SegmentType>::SegmentUserInformation
+SegmentManager<SegmentType>::getSegmentInformationWithReadAccessForUser(const posix::PosixUser& user) noexcept
+{
+    auto groupContainer = user.getGroups();
+
+    SegmentUserInformation segmentInfo{cxx::nullopt_t(), 0U};
+
+    // with the groups we can search for the readable segment of this user
+    for (const auto& groupID : groupContainer)
+    {
+        for (auto& segment : m_segmentContainer)
+        {
+            if (segment.getReaderGroup() == groupID)
+            {
+                segmentInfo.m_memoryManager = segment.getMemoryManager();
+                segmentInfo.m_segmentID = segment.getSegmentId();
+                return segmentInfo;
```

Comment:
So we return the "first" one if there is any and essentially none otherwise, OK.

So after this call we clearly know whether there is none.